core/site_battery.go

```diff
@@ -32,7 +32,7 @@ func (site *Site) SetBatteryMode(batMode api.BatteryMode) {
 
 func (site *Site) determineBatteryMode(loadpoints []loadpoint.API, smartCostActive bool) api.BatteryMode {
 	for _, lp := range loadpoints {
-		if lp.GetStatus() == api.StatusC && (smartCostActive || lp.GetMode() == api.ModeNow || lp.GetPlanActive()) {
+		if lp.GetStatus() == api.StatusC && (smartCostActive || lp.GetMode() == api.ModeNow || lp.GetPlanActive() || lp.GetMinSocNotReached()) {
```

Comment:
As `GetPlanActive()` is currently not used otherwise, that would be possible.

Isn't the API name `IsFastCharging()` a bit misleading, as the term "Fast charging" also includes other conditions, like `api.ModeNow`?

Reply:
Including that wouldn‘t hurt either?

Reply:
I also think that `IsFastCharging` is not precise enough. Would be more like a `IsFastChargingIn(Min)PvMode` function that includes `smartCostActive`, `GetPlanActive` and `GetMinSocNotReached`.

Other suggestion for better readability and consistancy:

```suggestion
		if lp.GetStatus() == api.StatusC && (lp.GetMode() == api.ModeNow || smartCostActive || lp.GetPlanActive() || lp.IsMinSocActive()) {
```

This sorting would fit my brain better, and we'd always use the word `active` for the exception-overwrite cases.

Reply:
We can simply include Now in the function

Reply:
> We can simply include Now in the function

How about this:

```suggestion
		if lp.IsFastChargingActive() || smartCostActive {
```

With `IsFastChargingActive()` implemented as:


```go
// IsFastChargingActive indicates if fast charging with maximum power is active
func (lp *Loadpoint) IsFastChargingActive() bool {
	lp.Lock()
	defer lp.Unlock()

	return lp.GetStatus() == api.StatusC &&
		(lp.GetMode() == api.ModeNow || lp.planActive || lp.minSocNotReached())
}
```

Note: That would shift nearly the whole logic in this API service. Following we should also add test case for that API?

Reply:
And: MUST NOT call public api (`lp.GetStatus`) while holding lock! This will likely deadlock depending on api.

Reply:
Oops, thanks for your hints. Then it should be better like this:
```suggestion
		if lp.GetStatus() == api.StatusC && (smartCostActive || lp.IsFastChargingActive()) {
```

And without public API call in `IsFastChargingActive()`:

```go
// IsFastChargingActive indicates if fast charging with maximum power is active
func (lp *Loadpoint) IsFastChargingActive() bool {
	lp.Lock()
	defer lp.Unlock()

	return lp.mode == api.ModeNow || lp.planActive || lp.minSocNotReached()
}

```

Reply:
LGMT- aber bitte RLock, Du liest ja nur.

Reply:
Danke. RLock ist drin.
Habe den Test auch noch um `smartCostActive` erweitert.